sunpy/map/mapbase.py

```diff
@@ -1542,20 +1542,13 @@ def resample(self, dimensions: u.pixel, method='linear'):
         new_meta = self.meta.copy()
 
         # Update metadata
+        for key in {'cdelt1', 'cd1_1', 'cd2_1'} & self.meta.keys():
+            new_meta[key] *= scale_factor_x
+        for key in {'cdelt2', 'cd1_2', 'cd2_2'} & self.meta.keys():
+            new_meta[key] *= scale_factor_y
         if 'pc1_1' in self.meta:
-            new_meta['pc1_1'] *= scale_factor_x
-            new_meta['pc2_1'] *= scale_factor_x
-            new_meta['pc1_2'] *= scale_factor_y
-            new_meta['pc2_2'] *= scale_factor_y
-        if 'cd1_1' in self.meta:
-            new_meta['cd1_1'] *= scale_factor_x
-            new_meta['cd2_1'] *= scale_factor_x
-            new_meta['cd1_2'] *= scale_factor_y
-            new_meta['cd2_2'] *= scale_factor_y
-        if 'cd1_1' not in self.meta and 'pc1_1' not in self.meta:
-            # Using the CROTA2 and CDELT formalism
-            new_meta['cdelt1'] *= scale_factor_x
-            new_meta['cdelt2'] *= scale_factor_y
+            new_meta['pc2_1'] *= scale_factor_x / scale_factor_y
+            new_meta['pc1_2'] *= scale_factor_y / scale_factor_x
```

Comment:
To make it clearer that the CDELTi keywords are used in conjunction with the PCi_j keywords, and not the CDi_j keywords when interpreting the metadata, perhpas it would be better to move the code for scaling CDELTi into this `if` block?

Reply:
As far as I can tell, there is no prohibition to have CDELTi when using CDi_j, even though the CDELTi will not be used for the transformation.  The header might be keeping them around to facilitate conversion back to a specific PCi_j formulation.  My thinking is that there is no reason to skip scaling CDELTi.